lang/java/reef-io/src/main/java/org/apache/reef/io/network/group/impl/driver/CommunicationGroupDriverImpl.java

```diff
@@ -133,17 +135,51 @@ public CommunicationGroupDriver addReduce(final Class<? extends Name<String>> op
     if (finalised) {
       throw new IllegalStateException("Can't add more operators to a finalised spec");
     }
-    LOG.finer(getQualifiedName() + "Adding reduce operator to tree topology: " + spec);
+    LOG.finer(getQualifiedName() + "Adding reduce operator to topology: " + spec);
     operatorSpecs.put(operatorName, spec);
-    final Topology topology = new TreeTopology(senderStage, groupName, operatorName, driverId, numberOfTasks, fanOut);
```

Comment:
Oh yes, there are two reasons:
1) The current Scatter operation (this PR)  only supports `FlatTopology`.
2) When a job uses two operations and the operations use different topology classes, the job fails (or rather, the job stalls on something).

Thus, it would be better to use only one type of topology class for all operations, and that's why I changed the topology of reduce and broadcast.